File: 4216-weighted-word-mapping/weighted-word-mapping.go
func mapWordWeights(words []string, weights []int) string {
    var res strings.Builder
    for _, word := range words {
        sum := 0
        for _, ch := range word {
            sum += weights[ch - 'a']
        }
        sum %= 26
        res.WriteByte(byte('z' - sum))
    }
    return res.String()
}

File: 4216-weighted-word-mapping/weighted-word-mapping.java
class Solution {
    public String mapWordWeights(String[] words, int[] weights) {
        String res = "";
        for(String word: words) {
            int sum = 0;
            for(char ch: word.toCharArray()) {
                int idx = ch - 'a';
                sum += weights[idx];
            }
            sum = sum % 26;
            res += (char)('z' - sum);
        }
        return res;
    }
}